### PR TITLE
Add library go tools

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @tnozicka @soltysh @deads2k

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+# $2 - input dirs
+# $3 - prefix
+# $4 - pkg
+# $5 - output
+# It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+
+clean:
+	$(RM) ./must-gather
```
Comment (soltysh) [inline Makefile]: That needs to be `openshift-dev-helpers` and `openshift-must-gather`

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+# $1 - target suffix
+# $2 - input dirs
+# $3 - prefix
+# $4 - pkg
+# $5 - output
+# It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+
+clean:
```
Comment (tnozicka) [inline Makefile]: drop clean, you'll get it with the multiple binary patch automatically

File: glide.yaml [hunk 3]
```diff
@@ -12,3 +12,6 @@ import:
   version: kubernetes-1.12.4
 - package: github.com/openshift/api
   version: master
+- package: github.com/openshift/library-go
+  repo:    https://github.com/soltysh/library-go
+  version: rebase-1.12.2
```
Comment (tnozicka) [inline glide.yaml]: setup your editor to always put "\n" at the end of the file :)

File: Makefile [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
+# This will call a macro called "add-bindata" which will generate bindata specific targets based on the parameters:
+# $0 - macro name
+# $1 - target suffix
+# $2 - input dirs
+# $3 - prefix
+# $4 - pkg
+# $5 - output
+# It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
```
Comment (tnozicka) [inline Makefile]: I don't think this repo uses bindata

**Comment by tnozicka** [comment]
@juanvallejo when you are making a dependency bump the glide files should be in a separate commit and then the vendor update goes into commit named `bump(*)`. Makes it easier to review - you check the glide files and CI checks that the vendor content matches what you specified in the glide. You optionally review the bump content based on circumstances.

File: glide.yaml (hunk 3 above)
Comment (juanvallejo) [reply glide.yaml]: :) will do

File: Makefile (hunk 4 above)
Comment (juanvallejo) [reply Makefile]: Thanks, removed

**Comment by juanvallejo** [comment]
@tnozicka thanks. Went ahead and split the bump commit into two separate ones. Renamed commit updating vendored deps to `bump(*)`. Additional comments also addressed

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
+WORKDIR /go/src/github.com/openshift/must-gather
+COPY . .
+ENV GO_PACKAGE github.com/openshift/must-gather
+RUN go build -ldflags "-X $GO_PACKAGE/pkg/version.versionFromGit=$(git describe --long --tags --abbrev=7 --match 'v[0-9]*')" ./cmd/openshift-must-gather
+
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+RUN mkdir -p /usr/share/bootkube/manifests
```
Comment (tnozicka) [inline Dockerfile]: why we need this line?

**Comment by juanvallejo** [comment]
> should we wait for rebase and vendoring library-go from the proper branch before merging?

@soltysh I'll defer this to you. I am currently using the 1.12 library-go branch from your repo, as discussed.

File: Dockerfile (hunk 5 above)
Comment (juanvallejo) [reply Dockerfile]: Sorry, leftover from copying the existing Makefile from another repo. Removed

**Comment by soltysh** [comment]
> > should we wait for rebase and vendoring library-go from the proper branch before merging?
> 
> @soltysh I'll defer this to you. I am currently using the 1.12 library-go branch from your repo, as discussed.

Yeah it's fine, although we still need to wait for https://github.com/openshift/library-go/pull/200 to land before merging this.

**Comment by tnozicka** [comment]
I am fine using @soltysh fork if it's based on merged PR

**Comment by juanvallejo** [comment]
> I am fine using @soltysh fork if it's based on merged PR

@tnozicka it's based on https://github.com/soltysh/library-go/tree/rebase-1.12.2

**Comment by soltysh** [comment]
@juanvallejo you can rebase on top of latest code, I've already picked Tomes's code.

**Comment by juanvallejo** [comment]
@soltysh 
> @juanvallejo you can rebase on top of latest code, I've already picked Tomes's code.

Thanks, done

**Pull request merged**